pyccl/emulators/baccoemu_linear_pk.py

```diff
@@ -0,0 +1,118 @@
+__all__ = ("BaccoemuLinear",)
+
+import numpy as np
+
+from .. import Pk2D
+from . import EmulatorPk
+
+
+class BaccoemuLinear(EmulatorPk):
+    """ Linear power spectrum emulator from baccoemu
+
+    This is an emulator of class linear Pk as a function of 8 cosmological
+    parameters (omega_cold, omega_baryons, A_s or sigma8_cold, ns, h, Mnu,
+    w0, wa) and the expansion factor.
+
+    See https://arxiv.org/abs/2104.14568
+    and https://bacco.dipc.org/emulator.html
+    """
+    def __init__(self):
+        # avoid tensorflow warnings
+        import warnings
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore', category=UserWarning)
+            import baccoemu
+            self.mpk = baccoemu.Matter_powerspectrum()
+        self.a_min = self.mpk.emulator['linear']['bounds'][-1][0]
+        self.a_max = self.mpk.emulator['linear']['bounds'][-1][1]
+        self.k_min = self.mpk.emulator['linear']['k'][0]
+        self.k_max = self.mpk.emulator['linear']['k'][-1]
+
+    def __str__(self) -> str:
+        return """baccoemu linear Pk module,
+k_min,k_max = ({}, {}),
+a_min,a_max = ({}, {})""".format(
+            self.k_min, self.k_max, self.a_min, self.a_max)
+
+    def _sigma8tot_2_sigma8cold(self, emupars, sigma8tot):
+        """Use baccoemu to convert sigma8 total matter to sigma8 cdm+baryons
+        """
+        if hasattr(emupars['omega_cold'], '__len__'):
+            _emupars = {}
+            for pname in emupars:
+                _emupars[pname] = emupars[pname][0]
+        else:
+            _emupars = emupars
```

Comment:
Most of the lack of coverage comes from not testing what happens when you pass one single set of cosmological parameters and scale factor. Can you make a test somewhere that checks e.g. that `get_pk_at_a` returns what it should when you pass a single value for `a`?

Reply:
Do this for the other Pk emulators too if you can.

Reply:
Done! Except ee2 -- but that's another story